allennlp/common/file_utils.py

```diff
@@ -474,6 +493,8 @@ def __init__(
             sync=True,
             readahead=False,
             meminit=False,
+            readonly=read_only,
+            lock=not read_only,
```

Comment:
We still need the lock. A file that's opened read-only by one process might be opened read-write by another.

Reply:
You need write permissions to use the lock though.

Reply:
Hmm. We should at least make it safe to do this if you have permission to use the lock though. You would definitely expect safety if you yourself have one process that writes to this and another one that reads.

If someone else tries to read it at the same time, and their process crashes because they couldn't lock, then we did the best we could.

Reply:
Good point.

Reply:
Just updated.